src/charm.py

```diff
@@ -61,14 +67,16 @@ def __init__(self, *args) -> None:
             return
         self._container_name = self._service_name = "ausf"
         self._container = self.unit.get_container(self._container_name)
-        self._nrf_requires = NRFRequires(charm=self, relation_name="fiveg_nrf")
+        self._nrf_requires = NRFRequires(charm=self, relation_name=NRF_RELATION_NAME)
+        self._webui = SdcoreConfigRequires(charm=self, relation_name=SDCORE_CONFIG_RELATION_NAME)
         self.unit.set_ports(SBI_PORT)
-        self._certificates = TLSCertificatesRequiresV3(self, "certificates")
+        self._certificates = TLSCertificatesRequiresV3(self, TLS_RELATION_NAME)
         self._logging = LogForwarder(charm=self, relation_name=LOGGING_RELATION_NAME)
         self.framework.observe(self.on.ausf_pebble_ready, self._configure_ausf)
         self.framework.observe(self.on.update_status, self._configure_ausf)
         self.framework.observe(self.on.fiveg_nrf_relation_joined, self._configure_ausf)
         self.framework.observe(self._nrf_requires.on.nrf_available, self._configure_ausf)
+        self.framework.observe(self._webui.on.webui_url_available, self._configure_ausf)
```

Comment:
self.framework.observe(self.on.sdcore_config_relation_joined, self._configure_ausf)

I suggest to observe `sdcore_config_relation_joined`  event.

Reply:
When the relation is joined until webui URL become available, configure method will not be triggered. For the moment, it seems it is affecting our charm state. If we agree on not checking `sdcore_config_relation_joined`, we can also remove `database_relation_joined` and  `fiveg_nrf_relation_joined` observation. They also have no effect on charm state.  Until database_created and nrf_url_available, we can not go forward.

Reply:
Oh, that's right. We can end up with outdated status. 
Handler added.